### ISSUE TITLE
Crontab.xml job not added to cron_schedule table if schedule includes day value

### ISSUE THREAD

**Issue by nbennett25** [magento/magento2#30247]
### Preconditions (*)

1. Magento 2.3.3
2. PHP 7.3

### Steps to reproduce (*)

1. Create custom module with crontab.xml file
2. Create job in crontab.xml file with schedule of '0 10 * * *' 
3. Run bin/magento cron:run at 9:45 am

### Expected result (*)

1. Record for job added to cron_schedule table

### Actual result (*)

1. No record is added to cron_schedule table

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround

**Comment by nbennett25** [comment]
@magento any updates here?

**Comment by nwcasebolt** [comment]
I can confirm this is still an issue in 2.4.2-p2, and I don't see any reference to a fix in the release notes for 2.4.3.

To see a core example of this, review the two entries in vendor/magento/module-gift-card-account/etc/crontab.xml:

- magento_giftcardaccount_update_states is scheduled to run at `0 1 * * *`, and has no entries in cron_schedule.

- magento_giftcardaccount_generage_codes_pool is scheduled to run at `30 * * * *` and does have entries in cron_schedule.

I have also noted the same behavior across third-party and custom crontab.xml files installed on our site.

**Comment by hostep** [comment]
@nwcasebolt: I can't confirm this problem on Magento 2.4.2-p2

I took a look at the `backend_clean_cache` cronjob (because Magento OS doesn't come with that giftcard module) which has a cron schedule set as: `30 2 * * *`.
Grepping the `var/log/cron.log` file gives me expected results:
```
$ grep 'backend_clean_cache' var/log/cron.log
[2021-09-05 00:30:03] main.INFO: Cron Job backend_clean_cache is run [] []
[2021-09-05 00:30:03] main.INFO: Cron Job backend_clean_cache is successfully finished.
[2021-09-06 00:30:03] main.INFO: Cron Job backend_clean_cache is run [] []
[2021-09-06 00:30:04] main.INFO: Cron Job backend_clean_cache is successfully finished.
[2021-09-07 00:30:03] main.INFO: Cron Job backend_clean_cache is run [] []
[2021-09-07 00:30:03] main.INFO: Cron Job backend_clean_cache is successfully finished.
[2021-09-08 00:30:03] main.INFO: Cron Job backend_clean_cache is run [] []
[2021-09-08 00:30:03] main.INFO: Cron Job backend_clean_cache is successfully finished.
```

(notice that the log's timestamps are in UTC while we are in CEST which is 2 hours ahead)


Your testing might be wrong because the entry from the `cron_schedule` table might already have been removed, because successfully ran cronjobs only stay in there for 60 minutes I believe.
So the better way to investigate this, is by looking into your `var/log/cron.log` file since that one should keep a record of all cronjobs that ran in the past.
